module/zfs/zfs_ctldir.c

```diff
@@ -1029,12 +1024,12 @@ zfsctl_snapshot_unmount(char *snapname, int flags)
 	}
 	rw_exit(&zfs_snapshot_lock);
 
-	argv[2] = kmem_asprintf(SET_UNMOUNT_CMD,
-	    flags & MNT_FORCE ? "-f " : "", se->se_path);
-	zfsctl_snapshot_rele(se);
+	if (flags & MNT_FORCE)
+		argv[i++] = "-f";
+	argv[i] = se->se_path;
 	dprintf("unmount; path=%s\n", se->se_path);
```

Comment:
It seems there was a potential use after free in the old code. The new code fixes this as a side effect.